activerecord/lib/active_record/validations/uniqueness.rb

```diff
@@ -20,6 +20,8 @@ def validate_each(record, attribute, value)
         finder_class = find_finder_class_for(record)
         value = map_enum_attribute(finder_class, attribute, value)
 
+        return if record.persisted? && !validation_needed?(finder_class, record, attribute)
```

Comment:
this idea of `validation_needed?` which checks

```ruby
return true if attributes.any? { |attr| record.attribute_changed?(attr) || record.read_attribute(attr).nil? }
```

could be a signal that there's a useful general validation option missing, something like `only_dirty: true` such that

```ruby
validate :thing, presence: true, only_dirty: true
```

would short circuit the presence validator on thing unless record.attribute_changed?("thing") and prevent it from checking the if/unless conditionals and continuing to the validate_each itself etc just like allow_blank does

If that seems interesting, I'm up to make the PR